File: examples/oop.js
class Vehicle {
  constructor$1(size) {
    this.w = this.h = size;
  }
  constructor$2(width, height) {
    this.w = width;
    this.h = height;
  }
  constructor(...args$) {
    switch (args$.length) {
      case 1:
        return this.constructor$1(...args$);
      case 2:
        return this.constructor$2(...args$);
    }
  }
  move$1(angle) {
    this.x += cos(angle);
    this.y += sin(angle);
  }
  move$2(dx, dy) {
    this.x += dx;
    this.y += dy;
  }
  move(...args$) {
    switch (args$.length) {
      case 1:
        return this.move$1(...args$);
      case 2:
        return this.move$2(...args$);
    }
  }
}

class Car extends Vehicle {
  constructor() {
    super(24);
    this instanceof Vehicle;
  }
  move(dx, dy) {
    super.move(dx, dy);
    this.move(0.5);
  }
}

File: examples/oop.java
// Package and imports are ignored
package my_package;

import java.awt.*;

// Interfaces and abstract classes are ignored
interface Thing {
  void move(float angle);
  void move(int dx, int dy);
}
abstract class MyAbstract {
  void thing();
}

// Implements is ignored
class Vehicle implements Thing {
  int w, h;
  int x, y;
  
  // Method and constructor names that appear more than once are parsed in a special way
  Vehicle(int size) {
    w = h = size;
  }
  Vehicle(int width, int height) {
    w = width;
    h = height;
  }
  void move(float angle) {
    x += cos(angle);
    y += sin(angle);
  }
  void move(int dx, int dy) {
    x += dx;
    y += dy;
  }
}

class Car extends Vehicle {

  Car() {
    super(24);

    this instanceof Vehicle; // true
  }

  @Overrides
  void move(int dx, int dy) {
    super.move(dx, dy);

    move(0.5);
  }
}
